File: parse_schedule.js
const fs = require('fs');

const html = fs.readFileSync('schedule.html', 'utf-8');

const days = ["Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"];
const scheduleData = {
    Sunday: [], Monday: [], Tuesday: [], Wednesday: [], Thursday: [], Friday: [], Saturday: []
};

// Split table into columns
const tbodyMatch = html.match(/<tbody>([\s\S]*?)<\/tbody>/);
if (tbodyMatch) {
    const tbody = tbodyMatch[1];
    // This splits by </td><td> roughly, but let's be more precise
    const tds = tbody.split(/<td[^>]*>/);
    
    // tds[0] is everything before first <td...>, tds[1] is hours column, tds[2..8] are days
    for (let i = 2; i <= 8; i++) {
        const dayIndex = i - 2;
        const dayName = days[dayIndex];
        const tdHtml = tds[i];
        
        // Find all boxes
        const regexBox = /<div class="station-sch-box[^>]*>([\s\S]*?)<\/div>(?=\s*<div class="station-sch-box|\s*<\/td>)/g;
        // Wait, nested divs make regex hard.
        // Let's just find all items directly
        
        const itemRegex = /<span class="station-sch-time">([^<]*)<\/span><span class="station-sch-title">([^<]*)<\/span><\/a>(\s*<div class="weektable-alt">)?/g;
        
        let match;
        while ((match = itemRegex.exec(tdHtml)) !== null) {
            scheduleData[dayName].push({
                time: match[1].trim(),
                title: match[2].trim(),
                is_alt: match[3] ? true : false
            });
        }
    }
}

fs.writeFileSync('schedule.json', JSON.stringify(scheduleData, null, 2));
console.log("Parsed schedule.html and created schedule.json");

File: parse_schedule.py
import json
from bs4 import BeautifulSoup
import re

with open('schedule.html', 'r') as f:
    html = f.read()

soup = BeautifulSoup(html, 'html.parser')
table = soup.find('table', id='station-sch')

days = ["Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"]

schedule_data = {day: [] for day in days}

if table:
    tbody = table.find('tbody')
    tr = tbody.find('tr')
    tds = tr.find_all('td', recursive=False)
    
    # First TD is hours (skip), then 7 TDs for days
    for i, td in enumerate(tds[1:]): # skip the first
        day = days[i]
        
        boxes = td.find_all('div', class_=re.compile(r'station-sch-scheduled|station-sch-unscheduled'))
        for box in boxes:
            items_div = box.find('div', class_='schedule-items')
            if not items_div:
                # Check if there is an a tag directly in box
                a_tag = box.find('a')
                if a_tag and a_tag.find('span', class_='station-sch-time'):
                    time = a_tag.find('span', class_='station-sch-time').text.strip()
                    title_span = a_tag.find('span', class_='station-sch-title')
                    title = title_span.text.strip() if title_span else ""
                    link = a_tag.get('href', '')
                    schedule_data[day].append({
                        "time": time,
                        "title": title,
                        "link": link,
                        "is_alt": False
                    })
                continue
                
            # Multiple items possible (main + alt)
            item_divs = items_div.find_all('div', recursive=False)
            if item_divs:
                for item in item_divs:
                    a_tag = item.find('a')
                    if a_tag:
                        time_span = a_tag.find('span', class_='station-sch-time')
                        time = time_span.text.strip() if time_span else ""
                        title_span = a_tag.find('span', class_='station-sch-title')
                        title = title_span.text.strip() if title_span else ""
                        link = a_tag.get('href', '')
                        is_alt = item.find('div', class_='weektable-alt') is not None
                        schedule_data[day].append({
                            "time": time,
                            "title": title,
                            "link": link,
                            "is_alt": is_alt
                        })
            else:
                # Direct a tags in schedule-items
                a_tags = items_div.find_all('a', recursive=False)
                for a_tag in a_tags:
                    time_span = a_tag.find('span', class_='station-sch-time')
                    time = time_span.text.strip() if time_span else ""
                    title_span = a_tag.find('span', class_='station-sch-title')
                    title = title_span.text.strip() if title_span else ""
                    link = a_tag.get('href', '')
                    schedule_data[day].append({
                        "time": time,
                        "title": title,
                        "link": link,
                        "is_alt": False
                    })

with open('schedule.json', 'w') as f:
    json.dump(schedule_data, f, indent=2)

print("Parsed schedule.html and created schedule.json")
